feat(treesitter): show query files in healthcheck

HarshK97 opened the pull request:
Related and fixes #36986

Problem:
The `:checkhealth vim.treesitter` command shows available parsers but doesn't show which query files are available for each language. This makes it difficult for users to see what treesitter functionality (highlights, injections, folds, etc.) is available.

Solution:
Add a new 'Treesitter queries' section that lists all query files grouped by language, showing the path to each query file.

HarshK97 committed:
feat(treesitter): show query files in healthcheck

Problem:
The :checkhealth vim.treesitter command shows available parsers but
doesn't show which query files are available for each language. This
makes it difficult for users to see what treesitter functionality
(highlights, injections, folds, etc.) is available.

Solution:
Add a new 'Treesitter queries' section that lists all query files
grouped by language, showing the path to each query file and marking
duplicates from different runtime paths as '(not loaded)'.

clason commented on runtime/lua/vim/treesitter/health.lua:
```diff
@@ -64,6 +64,53 @@ function M.check()
+        local is_duplicate = i > 1 and queries[i - 1].type == query.type
+        if is_duplicate then
+          health.ok(
+            string.format('Language: %-15s %s (not loaded): %s', lang, query.type, query.path)
```
...you are not using an LLM to (help) write these comments, are you?

HarshK97 replied:
yes i am, cause english is not my native and dont wanna have mistakes in them, which may mistakenly give wrong idea what i am talking

HarshK97 replied:
Thank you and I will make sure to not use LLM for answering here any more. Its not that language is barrier I can fully understand but just slowly.

and about validating queries, I will keep it simple right now but would love to learn how the queries are actually validated if you could maybe link me the files I should keep an eye on.

Ok from what I have gathered 
- I have to remove (Not loaded) marker and list all query files
- Preserve the runtimepath order
- remove the warning for not finding any query files
- remove the filename from path, as the type already shows what query is loaded

clason commented on runtime/lua/vim/treesitter/health.lua:
```diff
@@ -64,6 +64,53 @@ function M.check()
+      table.insert(queries_by_lang[lang], {
+        lang = lang,
+        type = query_type,
+        path = query_file,
```
This list is going to be very unwieldy; I would recommend stripping the filename  (since it's redundant with the leading `type:`).

HarshK97 committed:
fix(treesitter): addressing review feedback for query healthcheck

- Removed (not loaded) marker as queries can be extend using '; extends'
- List all query files in runtimepath order instead of old logic of not
loading duplicate queries
- Removed warning for empty queries
- Strip filename from the path as it is reduntant with query type

clason commented:
A cosmetic suggestion for better readability: Can you align the query types in a separate column as you do the language names? (The longest name is probably going to be `textobjects`.)

HarshK97 committed:
fix(treesitter): removed "Language" label for queries

HarshK97 committed:
fix(treesitter): cosmetic changes

- removed "Language" label for each query
- aligning "query.type" in separate column
